core/arch/arm/sm/sm_a32.S

```diff
@@ -216,6 +216,28 @@ FUNC sm_init , :
 	mrs	r1, cpsr
 	cps	#CPSR_MODE_MON
 	mov	sp, r0
+
```

Comment:
Yes I agree with you but it has to be handled in secure monitor which lies in OP-TEE for ARMv7. You can also refer to the link I have mentioned here.